frontend/helpers/generateFakeVault.ts

```diff
@@ -0,0 +1,132 @@
+    const id = faker.datatype.number();
+    const address = faker.finance.ethereumAddress();
+    const collateralType = faker.helpers.randomize(Object.keys(COLLATERALS));
+    const network = 'mainnet';
+    const lastSyncedAt = faker.date.recent();
+
+    return {
+        id,
+        address,
+        collateralType,
+        network,
+        lastSyncedAt,
+    };
+};
+
+const generateFakeVaultAmount = function (): VaultAmount {
+    const initialDebtDai = new BigNumber(faker.finance.amount());
+    const collateralAmount = new BigNumber(faker.finance.amount());
+
+    return {
+        initialDebtDai,
+        collateralAmount,
+    };
+};
+
+const generateFakerLiquidationLimits = function (): LiquidationLimits {
+    const maximumProtocolDebtDai = new BigNumber(faker.finance.amount());
+    const currentProtocolDebtDai = maximumProtocolDebtDai.dividedBy(faker.datatype.number({ max: 5 }));
+    const maximumCollateralDebtDai = new BigNumber(faker.finance.amount());
+    const currentCollateralDebtDai = maximumCollateralDebtDai.dividedBy(faker.datatype.number({ max: 5 }));
+
+    return {
+        maximumProtocolDebtDai,
+        currentProtocolDebtDai,
+        maximumCollateralDebtDai,
+        currentCollateralDebtDai,
+    };
+};
+
+export const generateFakeVault = function (): Vault {
+    const vaultBase = generateFakeVaultBase();
+    const vaultAmount = generateFakeVaultAmount();
+    const liquidationLimits = generateFakerLiquidationLimits();
+
+    return {
+        ...vaultBase,
+        ...vaultAmount,
+        ...liquidationLimits,
+    };
+};
+
+const generateFakeVaultTransactionFees = function (): VaultTransactionFees {
+    const transactionFeeLiquidationEth = new BigNumber(faker.datatype.number(0.5));
+    const transactionFeeLiquidationDai = transactionFeeLiquidationEth.multipliedBy(1600);
+
+    return {
+        transactionFeeLiquidationEth,
+        transactionFeeLiquidationDai,
+    };
+};
+
+const generateFakeOraclePrices = function (): OraclePrices {
+    const currentUnitPrice = new BigNumber(faker.datatype.number(2));
+    const nextUnitPrice = currentUnitPrice.multipliedBy(1.25);
+    const nextPriceChange = faker.date.soon();
+
+    return {
+        currentUnitPrice,
+        nextPriceChange,
+        nextUnitPrice,
+    };
+};
+
+export const generateFakeVaultTransaction = function (): VaultTransaction {
+    const fakeVault = generateFakeVault();
+    const fakeTransactionFees = generateFakeVaultTransactionFees();
+    const fakeOraclePrices = generateFakeOraclePrices();
+
+    const liquidationRatio = faker.datatype.number({ min: 110, max: 150 });
+    const minUnitPrice = faker.datatype.number();
+    const collateralizationRatio = fakeVault.collateralAmount.multipliedBy(minUnitPrice).toNumber();
+    const proximityToLiquidation = liquidationRatio - collateralizationRatio;
+
+    const state = proximityToLiquidation < 0 ? 'liquidatable' : 'not-liquidateable';
```

Comment:
There is also `liquidated` state

Reply:
I was not sure how and where I would get the:

- Transaction Hash
- New CollateralAuction info (is it just `collateralType` and `id` or does it change?

from a liquidated vault. Would we need to fetch it separately (like with finished collateral auctions) or how do we plan to integrate this?

Reply:
Very good points. I suggest we add `auctionId` that only exist on the liquidated vault? I imagine this info would be possible to fetch from the clip `Kick` event even after the vault had been liquidated. There should also be the transaction hash and the date of the liquidation based on the same event.

@KirillDogadin-std please suggest changes to the types in this or a separate PR? Or directly via the suggestion feature?

Reply:
#453